src/grain/harvest.js

```diff
@@ -0,0 +1,220 @@
+// @flow
+
+/**
+ * In SourceCred, projects regularly distribute Grain to contributors based on
+ * their Cred scores. This is called a "Harvest".
```

Comment:
Yea that makes sense. Don't think there's a clear right answer here, but I definitely understand your perspective now. Here's my perspective on this:

The word "distribution" has a lot of other connotation especially in code (e..g probability distribution, the current global distribution of grain, etc), when in reality its supposed to represent a single "distribution event", an act of doing a distribution.
![image](https://user-images.githubusercontent.com/7143583/79057625-92af9680-7c20-11ea-996d-985abe92f5d5.png)

With "payout", there's a similar problem, it has extra connotation that doesn't really fit what this "distribution event" is supposed to be (usually means its a payment of money to a single person, e.g. insurance payout).

The way I think about harvest is a discrete event that occurs on regular intervals (like on farm land), and the amount of the harvest is determined by how much land the community has to grow stuff on and how much fertilizer (investment from protocol labs) is used, and theirs split up among the workers of the farm based on how much effort they contributed to make it happen (cred being the "water" that flowed through the land). The word "harvest" to me implies a discrete event that took place while distribution does not. "Payout" also implies a discrete event to me.

>  Harvest implies Grain sort of grows/appreciates on it's own and we're simply taking that innate growth and moving it into your account.
I would actually argue that the word "grain" is what implies something that grows. Harvest  leans more into the whole farming analogy so it might accentuate that idea, but the idea of "grain growing" remains even if we don't use "harvest".

IMO "harvest" being something that people have to take time to understand is a feature not a bug. Someone could read "distribution" or "payout" and think they know what its supposed to be when they might actually have the wrong idea. Harvest also has more semantic meaning in relation to grain which might actually be more intuitive to some people vs "distribution" or "payout".

I agree that harvest (noun) vs harvest (verb) is a downside, but people taking extra time to understand "harvest" since its not something obvious just by its name might mitigate that. Another way to clarify the distinction would be to call the function "calculateHarvest", and everywhere else harvest is an object which implies noun.

Reply:
I started responding point-by-point on the merits of the terms, but realized harvesting seems like a meme more than it is a metaphor. Memes have their uses, but do want to be sure it's worth needing to teach the meaning. And we should probably take a step back to see what we'd want the term to do for us, what the goals are for it.

Generally speaking, I think some of the the reasons to use a metaphor are:
- As an association one can use to guess / intuit "mostly right" how something works.
- To use as a mental image / mnemonic / framework to remember things.
- To create a meme.

I don't think harvesting scores well on the first two points. As you mention it would require educating what it's supposed to mean. When guessing, for every correct option you can come up with 5 sane but wrong alternatives that fit the metaphor too. For example:

> and the amount of the harvest is determined by how much land the community has to grow stuff on and how much fertilizer (investment from protocol labs) is used, [...]

This landmass / fertilizer could be total absolute Cred, but it isn't. It could be the Grain bonding curve reserve ratio, but isn't. Confusingly, they're incorrect but not bad guesses, because they would be sane strategies to explore too. It's just not what the current implementation does.

---

> IMO "harvest" being something that people have to take time to understand is a feature not a bug. Someone could read "distribution" or "payout" and think they know what its supposed to be when they might actually have the wrong idea.

I think this is the key point of the discussion. If we agree that building a solid understanding of what happens under-the-hood is the goal here, the chosen term isn't the only part to consider.

For this goal, I think using a purposely vague term to force people to read the docs is not a robust way to go about it. It looks like a metaphor, so people are going to guess and not read the docs. Just like they're going to assume about distribution / payout without digging into the docs first.

I'm more confident that we can encourage this understanding by:
- Making sure we have well-written docs.
- Adding links to it all over the place.
(Such as your grain transaction records, where you might first encounter it)

But going back to the terms with this goal in mind. I think distribution has a leg up on harvesting, in that it makes asking important questions more intuitive. Because it doesn't give you any room to fill them in with assumptions.

- Why do we distribute grain? :shrug:
- How do we distribute grain? :shrug:
- Where does the grain to distribute come from? :shrug:

Compared to harvest, we can be off to a bad start:
- Why? _We probably "cultivated" it with a reason, so now is just there to take?_
- How? _We're probably collecting it somehow. As you do when harvesting._
- Where does it come from? _Probably we created the conditions to grow / emerge?_

Having a void there I think is preferable, because getting stumped on a question like that would be a great time to consult the docs :]

Reply:
> harvesting seems like a meme more than it is a metaphor. 

In this context, can you explain the difference between a meme and a metaphor? Genuinely curious.

Overall, I think you make reasonable arguments in favor of distribution over harvest, and I am not too attached either way, so I'm fine going with 'Distribute' and 'GrainDistribution' over harvest.

Reply:
> In this context, can you explain the difference between a meme and a metaphor? Genuinely curious.

A metaphor is about suggesting a similarity / analogy.
A meme is about catching on / widespread use / proliferating.

It's not like those are mutually exclusive so what I meant to say was:
It seems to me, the meme aspect is the main way harvesting is expected to create a collective understanding of the mechanism we're trying to give a name to. And the metaphor supports it as a meme, by being witty and visual and easy to remember. More than being an accurate comparison to something you already know, or mental model to remember the mechanism's inner details by.

So to teach the mechanism I think it's leaning towards a meme.

Don't get me wrong, I love a good meme. They can be fun and effective. Dramatic backstories and visuals on [Champions and Heroes](https://discourse.sourcecred.io/t/about-champions-and-heroes/291) was definitely for the memes, and you won't believe the smug face I had when half the online community for Katana's seemed to know what [Sneaky Amps](https://github.com/katana-dev/docs/blob/master/how-to/use-28-amp-types.md) were.

Reply:
Got it, thanks for the explanation. Marking this thread resolved as we are in agreement now on using the term Distribution.